### PR TITLE
docs: new upstream annotation format for peering

### PR DESCRIPTION
**Pull request by ndhanushkodi**
### Description
Adds new upstream annotation format

### PR Checklist

* [ ] updated test coverage
* [x] external facing docs updated
* [x] not a security concern

### CODE REVIEW

**Review by thisisnotashwin — APPROVED** [review]

File: website/content/docs/k8s/annotations-and-labels.mdx [hunk 1]
```diff
@@ -69,53 +69,102 @@ Resource annotations could be used on the Kubernetes pod to control connect-inje
...
+        The labeled format is required when using the cluster peering feature and specifying an upstream in another
+        peer. It also supports specifying Consul Enterprise Namespace, Partition, or Datacenter. Only one of peer,
+        datacenter, or partition can be specified.
+        - Service Name
+          To specify the upstream service, you can provide the name of the service, followed by `.svc`.
+          ```yaml
+          annotations:
+            "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc:[port]"
+          ```
+        - Peer or Datacenter
```
Comment (trujillo-adam) [inline website/content/docs/k8s/annotations-and-labels.mdx:125]:
```suggestion
        - Peer or datacenter: Place the peer or datacenter after `svc.` followed by either `peer` or `dc` and the port number.
```

File: website/content/docs/k8s/annotations-and-labels.mdx [hunk 2]
```diff
@@ -69,53 +69,102 @@ Resource annotations could be used on the Kubernetes pod to control connect-inje
...
+          ```yaml
+          annotations:
+            "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc.[service-peer].peer:[port]"
+          ```
+          ```yaml
+          annotations:
+            "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc.[service-dc].dc:[port]"
+          ```
+        - Consul Enterprise
+          You can specify a Consul Enterprise Namespace
```
Comment (trujillo-adam) [inline website/content/docs/k8s/annotations-and-labels.mdx:135]:
```suggestion
        - Namespace (required Consul Enterprise): Place the namespace after `svc.` followed by `ns` and the port number.
```

File: website/content/docs/k8s/annotations-and-labels.mdx [hunk 3]
```diff
@@ -69,53 +69,102 @@ Resource annotations could be used on the Kubernetes pod to control connect-inje
...
+          ```yaml
+          annotations:
+            "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc.[service-namespace].ns:[port]"
+          ```
+          When specifying a peer, datacenter or Consul Enterprise Admin Partition when namespaces are enabled, you must
```
Comment (trujillo-adam) [inline website/content/docs/k8s/annotations-and-labels.mdx:140]:
```suggestion
          When specifying a peer, datacenter, or admin partition when namespaces are enabled, you must
```

File: website/content/docs/k8s/annotations-and-labels.mdx [hunk 4]
```diff
@@ -69,53 +69,102 @@ Resource annotations could be used on the Kubernetes pod to control connect-inje
...
+          provide the namespace and one of (peer, datacenter, partition).
```
Comment (trujillo-adam) [inline website/content/docs/k8s/annotations-and-labels.mdx:141]:
```suggestion
          provide the namespace and a peer, datacenter, or partition.
```

File: website/content/docs/k8s/annotations-and-labels.mdx [hunk 5]
```diff
@@ -69,53 +69,102 @@ Resource annotations could be used on the Kubernetes pod to control connect-inje
...
+            "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc.[service-namespace].ns.[service-partition].ap:[port]"
+          ```
+          ```yaml
+          annotations:
+            "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc.[service-namespace].ns.[service-dc].dc:[port]"
+          ```
+
+      - Multiple Upstreams
+
+        If you would like to specify multiple services or upstreams, delimit them with commas. They can be of any of the Unlabeled, Labeled, or Prepared Query formats above when using the supported versions for the formats.
```
Comment (trujillo-adam) [inline website/content/docs/k8s/annotations-and-labels.mdx:157]:
```suggestion
      - Multiple upstreams: Delimit multiple services or upstreams with commas. You can specify any of the unlabeled, labeled, or prepared query formats when using the supported versions for the formats.
```

**Review by trujillo-adam — COMMENTED** [review]
Several suggestions to improve clarity and conform to the style guide

File: website/content/docs/k8s/annotations-and-labels.mdx [hunk 6]
```diff
@@ -69,53 +69,102 @@ Resource annotations could be used on the Kubernetes pod to control connect-inje
...
+
+          ~> **WARNING:** Setting a namespace when not using Consul Enterprise or using a version < 1.7
+          is not supported. It will be treated as part of the service name.
+
+        - Consul Enterprise Admin Partition
+          When using Consul Enterprise Admin Partitions [1.11+], your upstream services may be running in a different
+          partition. You must specify the namespace when specifying a partition, and the datacenter must be your local
+          datacenter if specified. Communicating across partitions using this method is only supported within a
+          datacenter. For cross partition communication across datacenters, see [cluster
+          peering](/docs/connect/cluster-peering/k8s).
```
[the remark this answers is not in the document]
Comment (ndhanushkodi) [reply website/content/docs/k8s/annotations-and-labels.mdx]: Yup!

File: website/content/docs/k8s/annotations-and-labels.mdx [hunk 7]
```diff
@@ -69,53 +69,102 @@ Resource annotations could be used on the Kubernetes pod to control connect-inje
...
-    ```yaml
-    annotations:
-      "consul.hashicorp.com/connect-service-upstreams":"[service-name]:[port]:[optional datacenter],[service-name]:[port]:[optional datacenter]"
-    ```
-
-    ```yaml
-    annotations:
-      "consul.hashicorp.com/connect-service-upstreams":"[service-name]:[port]:[optional datacenter],prepared_query:[query name]:[port]"
-    ```
+  - There are a few formats this annotation can take:
```
[the remark this answers is not in the document]
Comment (ndhanushkodi) [reply website/content/docs/k8s/annotations-and-labels.mdx]: I moved the indentation up and made that statement part of the previous paragraph, thanks!

[the remark this answers is not in the document]
Comment (ndhanushkodi) [reply website/content/docs/k8s/annotations-and-labels.mdx]: Yeah I agree it would be nice to organize this a bit better

**Comment by im2nguyen** [comment]
Merged this https://github.com/hashicorp/consul/pull/13550, which contains similar changes. For some reason, the run checks for this PR is stalling:

image

Going to close this PR as a result

**Pull request closed without merging**